chain/src/pipe.rs

```diff
@@ -336,6 +339,26 @@ fn validate_header(header: &BlockHeader, ctx: &mut BlockContext<'_>) -> Result<(
 		return Err(ErrorKind::InvalidBlockTime.into());
 	}
 
+	// We can determine output and kernel counts for this block based on mmr sizes from previous header.
+	// Assume 0 inputs and estimate a lower bound on the full block weight.
+	let num_outputs = header
+		.output_mmr_count()
+		.saturating_sub(prev.output_mmr_count());
+	let num_kernels = header
+		.kernel_mmr_count()
+		.saturating_sub(prev.kernel_mmr_count());
+
+	// Each block must contain at least 1 kernel and 1 output for the block reward.
+	if header.height > 0 && (num_outputs == 0 || num_kernels == 0) {
```

Comment:
validate_header is never called on genesis, so height test is redundant.
besides, even genesis has output and kernel:-)

Reply:
Good catch. I'll rework this and double check this assumption actually holds (even when running tests etc.)

Reply:
:+1: